lib/helpers/isAsyncFunction.js

```diff
@@ -1,11 +1,11 @@
 'use strict';
 
-const { inspect } = require('util');
+const primitiveTypes = new Set(['undefined', 'number', 'string', 'bigint', 'boolean', 'symbol']);
+const asyncFunctionPrototype = Object.getPrototypeOf(async function() {});
 
 module.exports = function isAsyncFunction(v) {
-  if (typeof v !== 'function') {
-    return;
+  if (primitiveTypes.has(typeof v)) {
```

Comment:
Why not just `if (typeof v !== 'function')`? I don't think we want to treat objects whose prototype is `asyncFunctionPrototype` as async functions...

Reply:
Yes you are right. improved ;)